web/src/pages/insights/HeaderCell.tsx

```diff
@@ -0,0 +1,245 @@
+import { Table as SUITable } from 'semantic-ui-react'
+import HtmlTooltip from '../../shared/utilities/htmlTooltip'
+import { ProjectInsightsDetails, ProjectInsightsSummary } from '../../sm-api'
+import FilterButton from './FilterButton'
+
+export type ColumnKey =
+    | keyof ProjectInsightsSummary
+    | keyof ProjectInsightsDetails
+    | 'mito'
+    | 'stripy'
+    | 'aligned_percentage'
+    | 'annotated_dataset_percentage'
+    | 'snv_index_percentage'
+    | 'sv_index_percentage'
+
+export interface HeaderCellConfig {
+    key: ColumnKey
+    label: string
+    sortable?: boolean
+    filterable?: boolean
+    tooltip?: string
+    className?: string
+}
+
+interface HeaderCellProps {
+    config: HeaderCellConfig
+    sortDirection?: 'ascending' | 'descending'
+    onSort: (key: ColumnKey, isMultiSort: boolean) => void
+    onFilter?: (key: ColumnKey, selectedOptions: string[]) => void
+    getUniqueOptionsForColumn: (key: ColumnKey) => string[]
+    selectedOptions: string[]
+}
+
+export const HeaderCell: React.FC<HeaderCellProps> = ({
+    config,
+    sortDirection,
+    onSort,
+    onFilter,
+    getUniqueOptionsForColumn,
+    selectedOptions,
+}) => {
+    const { key, label, sortable, filterable, tooltip, className } = config
+
+    return (
+        <SUITable.HeaderCell
+            className={`header-cell ${className || ''}`}
+            sorted={sortable ? sortDirection : undefined}
+            onClick={
+                sortable
+                    ? (event: React.MouseEvent<HTMLElement>) => onSort(key, event.shiftKey)
+                    : undefined
+            }
+            onMouseEnter={
+                className
+                    ? (event: React.MouseEvent<HTMLElement>) =>
+                          event.currentTarget.classList.add('expanded')
+                    : undefined
+            }
+            onMouseLeave={
+                className
+                    ? (event: React.MouseEvent<HTMLElement>) =>
+                          event.currentTarget.classList.remove('expanded')
+                    : undefined
+            }
+        >
+            {filterable && onFilter && (
+                <div className="filter-button">
+                    <FilterButton
+                        columnName={label}
+                        options={getUniqueOptionsForColumn(key)}
+                        selectedOptions={selectedOptions}
+                        onSelectionChange={(selectedOptions) => onFilter(key, selectedOptions)}
+                    />
+                </div>
+            )}
+            <div className="header-text">
+                {tooltip ? (
+                    <HtmlTooltip title={<p>{tooltip}</p>}>
+                        <div>{label}</div>
+                    </HtmlTooltip>
+                ) : (
+                    label
+                )}
+            </div>
+        </SUITable.HeaderCell>
+    )
+}
+
+export const detailsTableHeaderCellConfigs: HeaderCellConfig[] = [
```

Comment:
Naming conventions are [a bit over the place](https://stackoverflow.com/questions/40291766/naming-convention-for-const-object-keys-in-es6), but I'd probably suggest using screaming snake case, to suggest it's a global constant (and ditto below)

```suggestion
export const DETAILS_TABLE_HEADER_CELL_CONFIGS: HeaderCellConfig[] = [
```